src/test/java/org/mule/test/http/functional/requester/proxy/HttpRequestProxyDynamicConfigTestCase.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.test.http.functional.requester.proxy;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.tck.http.TestProxyServer;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.test.http.functional.requester.AbstractHttpRequestTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HttpRequestProxyDynamicConfigTestCase extends AbstractHttpRequestTestCase {
```

Comment:
I'd add another test using dynamic proxy authentication which was one of the original reasons behind the feature request.